src/DuckUtils.cpp

```diff
@@ -128,7 +128,7 @@ int saveWifiCredentials(std::string ssid, std::string password) {
       err = DUCK_ERR_EEPROM_WRITE;
     }
   }
-  return DUCK_ERR_NONE;
+  return err;
```

Comment:
Good Catch! Make sure the DuckUtils unit tests still pass. For this you will need to run the test on one of the ESP32 boards as this method requires WiFi.